lib/workspace.js

```diff
@@ -217,20 +217,25 @@ module.exports = {
         // FIXME: Check we are in "uninitialized"
         let result = await sqldb.queryOneRowAsync(sql.select_workspace_paths, {workspace_id});
         const {course_path, qid} = result.rows[0];
+        result = await sqldb.queryOneRowAsync(sql.select_workspace_version, {workspace_id});
+        const { workspace_version } = result.rows[0];
         const localPath = `${course_path}/questions/${qid}/workspace`;
-        const s3Path = `${config.workspaceS3Bucket}/workspace-${workspace_id}`;
+
+        const s3Name = `workspace-${workspace_id}-${workspace_version}/current`;
+        const s3Path = `${config.workspaceS3Bucket}/${s3Name}`;
+
         const now = new Date().toISOString().replace(/[-T:.]/g, '-');
         const zipName = `workspace-${workspace_id}-${now}.zip`;
         const zipPath = path.join(config.workspaceMainZipsDirectory, zipName);
 
         debug(`Zipping ${localPath} as ${zipPath}`);
         await zipDirectory(localPath, zipPath);
         const isDirectory = false;
-        const s3ZipPath = `workspace-${workspace_id}/initial.zip`;
+        const s3ZipPath = s3Name.replace('current', 'initial.zip');
         await awsHelper.uploadToS3Async(zipPath, isDirectory, s3ZipPath, zipPath);
 
         debug(`Syncing ${localPath} to ${s3Path}`);
-        await module.exports.uploadDirectoryToS3Async(localPath, `${s3Path}/current`);
+        await module.exports.uploadDirectoryToS3Async(localPath, `${s3Path}`);
```

Comment:
Nit: remove unnecessary formatter:
```suggestion
        await module.exports.uploadDirectoryToS3Async(localPath, s3Path);
```